### DISCUSSION TITLE
Why am I not able to set the `height` of `.nav-footer-right`?

### DISCUSSION THREAD

**Discussion by dense-set** [quarto-dev/quarto-cli#6343] [Q&A] [answered]
### Description

So, I am creating a website project and a footer therein using the `page-footer` option's `left`, `right`, `center` suboptions. This creates the HTML like this:
```html
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">left</div>   
    <div class="nav-footer-center">center</div>
    <div class="nav-footer-right">right</div>
  </div>
</footer>
```

Just for the sake of it, I was fiddling with CSS and found that I am not able to set the `height` of `.nav-footer-left`, *etc.* contents using the percentage notation, like so:
```css
.nav-footer-left {
    height: 200%;
}
```

However, I *am* able to set it via pixels.

Why is it so? (I had a long look at the generated Bootsrap CSS files, but that didn't help.)

**Comment by cderv** [comment]
I would say pixel is an absolute unit, and % is a relative one. So 200% needs to apply on something and it could be limited too. 

Also know that `footer` is a Flex element 
https://github.com/quarto-dev/quarto-cli/blob/167c748ca99ad3ec2f4bc5414f887d9352c0fbd1/src/resources/projects/website/navigation/quarto-nav.scss#L763-L775

So you need to understand how that works (https://css-tricks.com/snippets/css/a-guide-to-flexbox/) or remove the flex CSS

Hope it helps

**Reply by mcanouil** [reply, marked as answer]
In addition, be aware that relatives units are to the parent element.
If the parent element has no absolute dimensions, the child element can't have relative dimensions.
Usually `body` has no height or weight.

This is why Quarto defined a big div inside body, in order to be able to have relative dimensions elements in it, for example to make grids.
